File: vben-admin/apps/web-antd/src/api/email-template.ts
import { requestClient } from '#/api/request';

export interface EmailTemplate {
  id: number;
  code: string;
  name: string;
  subject: string;
  content: string;
  variables: string;
  status: number;
  updated_at: string;
  created_at: string;
}

export function getEmailTemplatesApi() {
  return requestClient.get('/admin/email-templates');
}

export function saveEmailTemplateApi(data: { id: number; subject: string; content: string; status: number }) {
  return requestClient.post('/admin/email-templates/save', data);
}

export function previewEmailTemplateApi(code: string) {
  return requestClient.get('/admin/email-templates/preview', { params: { code } });
}

export function testEmailTemplateApi(code: string, test_to: string) {
  return requestClient.post('/admin/email-templates/test', { code, test_to });
}

File: go-api/internal/model/email_template.go
package model

// EmailTemplate 邮件模板
type EmailTemplate struct {
	ID        int    `json:"id"`
	Code      string `json:"code"`      // register / reset_password / system_notify
	Name      string `json:"name"`
	Subject   string `json:"subject"`   // 支持 {site_name} 等变量
	Content   string `json:"content"`   // HTML内容，支持变量
	Variables string `json:"variables"` // 可用变量列表(逗号分隔)
	Status    int    `json:"status"`    // 1=启用 0=禁用
	UpdatedAt string `json:"updated_at"`
	CreatedAt string `json:"created_at"`
}

// EmailTemplateSaveRequest 保存模板
type EmailTemplateSaveRequest struct {
	ID      int    `json:"id" binding:"required"`
	Subject string `json:"subject" binding:"required"`
	Content string `json:"content" binding:"required"`
	Status  int    `json:"status"`
}

// EmailTemplatePreviewRequest 预览/测试
type EmailTemplatePreviewRequest struct {
	Code   string `json:"code" binding:"required"`
	TestTo string `json:"test_to"` // 留空=仅预览，有值=发送测试
}
